mcs/class/System/Makefile

```diff
@@ -30,7 +30,7 @@ TEST_RESOURCES = \
 
 USE_XTEST_REMOTE_EXECUTOR = YES
 XTEST_LIB_REFS = System System.Core System.Net Facades/System.Threading.Tasks Facades/System.Runtime.InteropServices.RuntimeInformation System.Net.Http
-LIB_MCS_FLAGS = -d:CONFIGURATION_2_0 -d:XML_DEP -d:SECURITY_DEP $(REFERENCE_SOURCES_FLAGS) -unsafe $(RESOURCE_FILES:%=-resource:%) -nowarn:436
+LIB_MCS_FLAGS = -d:CONFIGURATION_2_0 -d:FEATURE_COMPILED -d:XML_DEP -d:SECURITY_DEP $(REFERENCE_SOURCES_FLAGS) -unsafe $(RESOURCE_FILES:%=-resource:%) -nowarn:436
```

Comment:
The Linux x64 failure is due to:

```
amd64/external/corefx/src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/CompiledRegexRunnerFactory.cs(15,45): error CS0246: The type or namespace name 'DynamicMethod' could not be found (are you missing a using directive or an assembly reference?)
/mnt/jenkins/workspace/test-mono-pull-request-amd64/external/corefx/src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/CompiledRegexRunnerFactory.cs(15,63): error CS0246: The type or namespace name 'DynamicMethod' could not be found (are you missing a using directive or an assembly reference?)
/mnt/jenkins/workspace/test-mono-pull-request-amd64/external/corefx/src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/CompiledRegexRunnerFactory.cs(15,88): error CS0246: The type or namespace name 'DynamicMethod' could not be found (are you missing a using directive or an assembly reference?)
/mnt/jenkins/workspace/test-mono-pull-request-amd64/external/corefx/src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/CompiledRegexRunnerFactory.cs(11,26): error CS0246: The type or namespace name 'DynamicMethod' could not be found (are you missing a using directive or an assembly reference?)
/mnt/jenkins/workspace/test-mono-pull-request-amd64/external/corefx/src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/CompiledRegexRunnerFactory.cs(12,26): error CS0246: The type or namespace name 'DynamicMethod' could not be found (are you missing a using directive or an assembly reference?)
/mnt/jenkins/workspace/test-mono-pull-request-amd64/external/corefx/src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/CompiledRegexRunnerFactory.cs(13,26): error CS0246: The type or namespace name 'DynamicMethod' could not be found (are you missing a using directive or an assembly reference?)
/mnt/jenkins/workspace/test-mono-pull-request-amd64/external/corefx/src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexLWCGCompiler.cs(54,18): error CS0246: The type or namespace name 'DynamicMethod' could not be found (are you missing a using directive or an assembly reference?)
../../build/library.make:342: recipe for target '../../class/lib/monotouch/System.dll' failed
make[8]: Leaving directory '/mnt/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/System'
make[8]: *** [../../class/lib/monotouch/System.dll] Error 1
```

We need to disable compilation on monotouch etc.

Reply:
@akoeplinger oh, I successfully compiled the PR on my linux but I forgot to add "d:FEATURE_COMPILED" there so I blamed CI, thanks for noticing!

Reply:
just compiling on Linux isn't enough to enable the additional profiles, you can use `--with-runtime_preset=all` during configure to build all the profiles.